core/src/autogluon/core/searcher/bayesopt/autogluon/searcher_factory.py

```diff
@@ -167,6 +168,63 @@ def _resource_for_acquisition(kwargs, hp_ranges_cs):
     return resource_for_acquisition
 
 
+def constrained_gp_fifo_searcher_factory(**kwargs) -> ConstrainedGPFIFOSearcher:
+    """
+    Creates ConstrainedGPFIFOSearcher object, based on kwargs equal to search_options
+    passed to and extended by scheduler (see FIFOScheduler).
+
+    :param kwargs: search_options coming from scheduler
+    :return: ConstrainedGPFIFOSearcher object
+
+    """
+    assert kwargs['scheduler'] == 'fifo', \
+        "This factory needs scheduler = 'fifo' (instead of '{}')".format(
+            kwargs['scheduler'])
+    # Common objects
+    hp_ranges_cs, random_seed, gpmodel, model_args, profiler, _map_reward, \
+    skip_optimization, debug_log = _create_common_objects(**kwargs)
+    constraint_metric_name = kwargs.get('constraint_metric', DEFAULT_CONSTRAINT_METRIC)
```

Comment:
These are internal names, no?

Of course, the user-facing names (appearing in train_fn) must be configurable.

Reply:
Done.